turbopack/crates/turbopack-ecmascript/src/simple_tree_shake/mod.rs

```diff
@@ -0,0 +1,86 @@
+//! Intermediate tree shaking that uses global information but not good as the full tree shaking.
+
+use anyhow::{Context, Result};
+use rustc_hash::{FxBuildHasher, FxHashMap, FxHashSet};
+use turbo_rcstr::RcStr;
+use turbo_tasks::{ResolvedVc, Vc};
+use turbopack_core::{module_graph::ModuleGraph, resolve::ExportUsage};
+
+use crate::chunk::EcmascriptChunkPlaceable;
+
+#[turbo_tasks::function]
+pub async fn get_module_export_usages(
+    graph: ResolvedVc<ModuleGraph>,
+    module: ResolvedVc<Box<dyn EcmascriptChunkPlaceable>>,
+) -> Result<Vc<ModuleExportUsageInfo>> {
+    let export_usage_info = compute_export_usage_info(graph)
+        .resolve_strongly_consistent()
+        .await?;
+
+    let export_usage_info = export_usage_info.await?;
+
+    let Some(exports) = export_usage_info.used_exports.get(&module) else {
+        // We exclude template files from tree shaking because they are entrypoints to the module
+        // graph.
+        return Ok(ModuleExportUsageInfo::all());
+    };
+
+    Ok(ModuleExportUsageInfo {
+        exports: exports.clone(),
+    }
+    .cell())
+}
+
+#[turbo_tasks::function(operation)]
+async fn compute_export_usage_info(graph: ResolvedVc<ModuleGraph>) -> Result<Vc<ExportUsageInfo>> {
+    let mut result = ExportUsageInfo::default();
+
+    graph
+        .await?
+        .traverse_all_edges_unordered(|(_, ref_data), target| {
+            if let Some(target_module) =
+                ResolvedVc::try_downcast::<Box<dyn EcmascriptChunkPlaceable>>(target.module)
+            {
+                result
+                    .used_exports
+                    .entry(target_module)
+                    .or_default()
+                    .insert(ref_data.export.clone());
+            }
+
+            Ok(())
+        })
+        .await
+        .context("failed to traverse module graph")?;
+
+    Ok(result.cell())
+}
+
+#[turbo_tasks::value]
+#[derive(Default)]
+pub struct ExportUsageInfo {
+    used_exports: FxHashMap<ResolvedVc<Box<dyn EcmascriptChunkPlaceable>>, FxHashSet<ExportUsage>>,
```

Comment:
This should instead be:

```suggestion
    used_exports: FxHashMap<ResolvedVc<Box<dyn EcmascriptChunkPlaceable>>, Vc<ModuleExportUsageInfo>,
```

so that it's a Vc already, so you don't need the clone above for `exports: exports.clone()`